src/Cli/dotnet/commands/dotnet-workload/install/IInstaller.cs

```diff
@@ -51,7 +51,7 @@ internal interface IInstaller : IWorkloadManifestInstaller
         /// <param name="manifestContents">The JSON contents describing the install state.</param>
         void SaveInstallStateManifestVersions(SdkFeatureBand sdkFeatureBand, Dictionary<string, string> manifestContents);
 
-        void UpdateInstallMode(SdkFeatureBand sdkFeatureBand, bool newMode);
+        void UpdateInstallMode(SdkFeatureBand sdkFeatureBand, bool? newMode);
```

Comment:
nb: correctness change - since the newMode can be 'unset', null is a reasonable way to represent that. See 'rollback' as an example of when mode might need to be explicitly unset.